api/src/test/java/org/bahmni/module/communication/service/impl/SMSTemplateServiceTest.java

```diff
@@ -0,0 +1,104 @@
+package org.bahmni.module.communication.service.impl;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.messagesource.MessageSourceService;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@PowerMockIgnore("javax.management.*")
+@PrepareForTest(Context.class)
+@RunWith(PowerMockRunner.class)
+public class SMSTemplateServiceTest {
+
+    @Mock
+    private AdministrationService administrationService;
+
+    @Mock
+    private MessageSourceService messageSourceService;
+
+    @InjectMocks
+    CommunicationServiceImpl communicationService;
+
+    private SMSTemplateService smsTemplateService;
+
+
+    @Before
+    public void setUp() {
+        PowerMockito.mockStatic(Context.class);
+        when(Context.getAdministrationService()).thenReturn(administrationService);
+        when(Context.getMessageSourceService()).thenReturn(messageSourceService);
+        smsTemplateService = new SMSTemplateService();
+    }
+
+    @Test
+    public void shouldReturnFormattedMessageWhenGlobalPropertyIsSet() {
+        String smsTemplate = "template.key";
+        String expectedTemplate = "Hello {name}, Your appointment is on {date}.";
+        Map<String, String> arguments = new HashMap<>();
+        arguments.put("name", "John");
+        arguments.put("date", "2024-03-15");
+        when(administrationService.getGlobalProperty(smsTemplate)).thenReturn(expectedTemplate);
+
+        String result = smsTemplateService.templateMessage(smsTemplate, arguments);
+
+        assertEquals("Hello John, Your appointment is on 2024-03-15.", result);
```

Comment:
Is the date format in backend YYYY-MM-DD, or is this based on what we put in "date" object? In general, in Bahmni we don't want to confuse people, and would prefer to send date in email as: "15-Mar-2024" so its unambiguous.

Reply:
The placeholder values sent to this module are Strings. The appointment module formats the Date object to string and the format is coming from [here](https://github.com/Bahmni/openmrs-module-appointments/blob/3b7fbc118e74e2eac8f2c3b2ac7d61f6c723dcd2/omod/src/main/resources/config.xml#L158) with default value being `dd-MM-yyyy' Time 'HH:mm`